src/main/java/panda/component/Storage.java

```diff
@@ -0,0 +1,95 @@
+package panda.component;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import panda.command.AlterMarkCommand;
+import panda.command.Command;
+import panda.command.NewTaskCommand;
+import panda.exception.CorruptedFileException;
+import panda.exception.PandaException;
+import panda.task.Deadline;
+import panda.task.Event;
+import panda.task.Todo;
+
+import java.util.ArrayList;
+
+public class Storage {
+    private File cacheFile;
+    
+    /**
+     * Constructs a new Storage.
+     * 
+     * @param filePath the path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        cacheFile = new File(filePath);
+        if (!cacheFile.exists()) {
+            try {
+                cacheFile.createNewFile();
+            } catch (IOException e) {
+                System.out.println("An error occurred.");
+                e.printStackTrace();
+            }   
+        }
+    }
+
+    /**
+     * Loads tasks from the file.
+     * 
+     * @return a list of commands representing the tasks loaded from the file.
+     * @throws PandaException if an error occurs during loading.
+     */
+    public ArrayList<Command> load() throws PandaException {
+        ArrayList<Command> clist = new ArrayList<>();
+        try {
+            Scanner myReader = new Scanner(cacheFile);
+            int idx = 1;
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                String[] parts = data.split("\\|");
+                for (int i = 0; i < parts.length; i++) {
+                    parts[i] = parts[i].trim();
+                }
+                if(parts[0].equals("T")) {
+                    clist.add(new NewTaskCommand(new Todo(parts[2])));
+                } else  {
+                    if(parts[0].equals("D")) {
```

Comment:
Perhaps use else if statements or switch statements to make the code more concise?